toolkit/ThreeWToolkit/data_visualization/plot_series.py

```diff
@@ -0,0 +1,62 @@
+from typing import cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+
+from .base_visualizer import BaseVisualizer
+
+
+class PlotSeries(BaseVisualizer):
+    def __init__(
+        self,
+        series: pd.Series,
+        title: str,
+        xlabel: str,
+        ylabel: str,
+        overlay_events: bool = False,
+        **plot_kwargs,
+    ) -> None:
+        self.series = series
+        self.title = title
+        self.xlabel = xlabel
+        self.ylabel = ylabel
+        self.overlay_events = overlay_events
+        self.plot_kwargs = plot_kwargs
+
+    def plot(self, ax: Axes | None = None) -> tuple[Figure, Axes]:
+        if ax is None:
+            fig, ax = plt.subplots(figsize=(12, 6))
+        else:
+            fig = cast(Figure, ax.get_figure())
+
+        ax.plot(
+            self.series.index,
+            np.asarray(self.series.values),
+            label="Value",
+            **self.plot_kwargs,
+        )
+        ax.set_title(self.title)
+        ax.set_xlabel(self.xlabel)
+        ax.set_ylabel(self.ylabel)
+        ax.grid(True, alpha=0.3)
+
+        if self.overlay_events:
+            nan_dates = self.series.index[self.series.isna()]
+            for date in nan_dates:
+                ax.axvline(x=date, color="red", linestyle="--", alpha=0.7, linewidth=1)
+
```

Comment:
Maybe is worth checking if the plot would be empty. If so, similarly as suggested for the other plots, we could raise an error here.